*Step 1*
**Understand the Task:-**
**We’ll use:
Titanic dataset**

**👉 Goal:
Predict → passenger survived or no**

In [38]:
import pandas as pd
df = pd.read_csv("/kaggle/input/datasets/yasserh/titanic-dataset/Titanic-Dataset.csv")
df["Sex"].unique()  #survived (1) or did not survive (0)

array(['male', 'female'], dtype=object)

**Step 2: Prepare Data**

In [39]:
##Select important columns
df = df[["Survived", "Pclass", "Sex", "Age"]]
# the "Pclass", "Sex", "Age" are the inputs for the target survived or not 

##Convert text → numbers: or Converted categorical data to numerical
df["Sex"] = df["Sex"].astype(str).str.lower().str.strip()
df["Sex"] = df["Sex"].map({"male": 0, "female": 1})
## Why?
# ML models cannot understand text
# So:
# male → 0
# female → 1

## Handle missing values:
df = df.fillna(df.mean(numeric_only=True))
# Why?
# Some Age values are missing 😕
# 👉 ML models cannot handle empty values

# So we fill with average (mean)
df.head()

,Survived,Pclass,Sex,Age
0,0,3,0,22.0
1,1,1,1,38.0
2,1,3,1,26.0
3,1,1,1,35.0
4,0,3,0,35.0


**Step 3: Split Data**

In [40]:
from sklearn.model_selection import train_test_split

X = df[["Pclass", "Sex", "Age"]]  # Inputs
y = df["Survived"]               # Output

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
#You divide data into:
# Training data (80%) → model learns from this
# Testing data (20%) → model is tested on this
# 👉 Why?
# Train model on some data
# Test on unseen data
X_train.shape, X_test.shape
#  X_train.head() , is used to see the training data , and it should hve Pcls,sex,Age
#y_train.head() # is used to check the target

((712, 3), (179, 3))

👉 Think like this:

X = what model uses to learn,
y = what model tries to predict
this:- train_test_split(...)  , is used to split the data
. What is test_size=0.2?
👉 Means:
80% → Training
20% → Testing

**Step 4: Train Model**

In [41]:
from sklearn.linear_model import LogisticRegression
#👉 You are importing a machine learning algorithm
# 📌 What is Logistic Regression?
# 👉 A model used for classification problems

model = LogisticRegression()## creating a model
#You are creating an empty brain 🧠
# At this point:
# ❌ It knows nothing
# ❌ No learning yet

model.fit(X_train, y_train)
# 🧠 What does .fit() do?
# 👉 It tells the model:
# “Look at this data and learn patterns

LogisticRegression()

**Step 5: Predict**

In [42]:
predictions = model.predict(X_test)
# 🧠 What is happening?
# 👉 Your trained model is now making predictions on new data
# Remember:
# It has never seen X_test before
# So this is a real test

#See first few predictions:
predictions[:10]

array([0, 1, 1, 0, 1, 0, 1, 0, 0, 0])

**🚀 🔹 STEP 6: Check Accuracy**

In [43]:
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, predictions)
print(accuracy)
# 🧠 What is happening?

# 👉 You are comparing:
# Actual answers → y_test
# Model answers → predictions

0.7877094972067039


**0.7877 → ~78.77%**
**👉 Meaning:
Your model correctly predicted ~79 out of 100 passengers**
You completed a full ML pipeline:

✔️ Data cleaning
✔️ Feature selection
✔️ Train-test split
✔️ Model training
✔️ Prediction
✔️ Evaluation